tests/many_to_many/tests.py

```diff
@@ -509,6 +509,24 @@ def test_clear(self):
         self.assertQuerysetEqual(self.a4.publications.all(), [])
         self.assertQuerysetEqual(self.p2.article_set.all(), ['<Article: NASA finds intelligent life on Earth>'])
 
+    def test_clear_after_prefetch(self):
```

Comment:
Can you add tests for `.set()` as well? I realize the implementation goes through the other methods now, but it's a bit tricky to follow and we don't want to lose that behavior in a future refactoring.